### PR TITLE
fix: make `all_goals` admit goals on failure

### PR DESCRIPTION
**Pull request by kmill**
New behavior: when in recovery mode, if any tactic fails in `all_goals` then the metacontext is restored and all goals are admitted. 

Without this, it can leave partially-solved metavariables and incomplete goal lists.

### CODE REVIEW

**Review by tobiasgrosser — APPROVED** [review]
Nice. This will be great to have in.

File: src/Init/Tactics.lean [hunk 1]
```diff
@@ -272,12 +272,20 @@ macro nextTk:"next " args:binderIdent* arrowTk:" => " tac:tacticSeq : tactic =>
   -- Limit ref variability for incrementality; see Note [Incremental Macros]
   withRef arrowTk `(tactic| case%$nextTk _ $args* =>%$arrowTk $tac)
 
-/-- `all_goals tac` runs `tac` on each goal, concatenating the resulting goals, if any. -/
+/--
+`all_goals tac` runs `tac` on each goal, concatenating the resulting goals.
+If the tactic fails on any goal, the entire `all_goals` tactic fails.
+
+This is equivalent to `any_goals try tac`.
```
Comment (nomeata) [inline src/Init/Tactics.lean:279]: Is it? This doesn't look like it would fail just when `tac` fails for one goal. (But its late, I might be wrong)

Comment (kmill) [reply src/Init/Tactics.lean:279]: Thanks, it's definitely not!

**Comment by tobiasgrosser** [comment]
Thank you, @kmill, for the quick turnaround.

**Pull request merged**